lib/tasks/integration.rake

```diff
@@ -1,144 +1,176 @@
+  banner = print_banner_around
 
-  banner = ->(line) do
-    puts
-    puts "=" * 40
-    puts "== #{Color::BOLD}#{line}#{Color::CLEAR}"
-    puts "=" * 40
-    puts
+  require 'ci_reporter_shell'
+  report = CiReporterShell.report('tmp/junit')
 
-    line
+  command = nil
+  result = report.execute(test_command, env: {RAILS_ENV: Rails.env}) do |cmd|
+    banner.call("BEGIN: #{cmd}")
+    command = cmd
   end
 
-  require 'ci_reporter_shell'
+  banner.call("FINISH (#{result.success? ? success : failure}): #{command} in #{format('%.1fs', result.time)}")
+
+  abort "#{test_command} FAILED" unless result.success?
+end
+
+desc 'The default execution: the whole CI suite'
+task :integrate => 'integrate:parallel'
 
-  report = CiReporterShell.report('tmp/junit')
 
-  junit = Dir['tmp/junit/**']
+namespace :integrate do
 
-  if junit.present?
-    puts 'WARNING: tmp/junit is not empty'
-    puts junit
-    abort 'Will not continue, tmp/junit is not empty'
+  test_commands.each_key do |command|
+    desc "Runs tests with #{command}"
+    task command.to_s => :prepare do
+      run_tests(test_commands[command])
+    end
   end
 
-  total_time = 0
+  desc 'Runs subset of full test suite, in parallel (1/8)'
+  task :parallel_1 => %i[verify_empty_reports_folder prepare cucumber_javascript rspec]
+
+  desc 'Runs subset of full test suite, in parallel (2/8)'
+  task :parallel_2 => %i[verify_empty_reports_folder prepare integration]
+
+  desc 'Runs subset of full test suite, in parallel (3/8)'
+  task :parallel_3 => %i[verify_empty_reports_folder prepare integration]
 
-  results = tasks.map do |task|
+  desc 'Runs subset of full test suite, in parallel (4/8)'
+  task :parallel_4 => %i[verify_empty_reports_folder prepare functional main_suite]
 
-    command = nil
-    result = report.execute(task, env: {RAILS_ENV: Rails.env}) do |cmd|
-      banner.("BEGIN: #{cmd}")
-      command = cmd
+  desc 'Runs subset of full test suite, in parallel (5/8)'
+  task :parallel_5 => %i[verify_empty_reports_folder prepare cucumber_non_tagged]
+
+  desc 'Runs subset of full test suite, in parallel (6/8)'
+  task :parallel_6 => %i[verify_empty_reports_folder prepare cucumber_for_categories]
+
+
+
+  desc 'Runs the whole continuous integration test suite'
+  task :parallel, [:log] do
+
+    banner = print_banner_around
+
+    time = Benchmark.measure do
+
+      Rake::Task.tasks.select { |task| task.name =~ /^integrate:parallel_/ }.map(&:invoke)
+      Rake::Task['integrate:license_checks'].invoke
     end
 
-    summary << banner.("FINISH (#{result.success? ? success : failure}): #{command} in #{'%.1fs' % result.time}")
-    total_time += result.time
+    # TODO: print summary
+    banner.call("Finished in #{format('%.1fs', time.real)}\n\t")
+
+    Rake::Task['integrate:report_coverage_to_codeclimate'].invoke(7)
 
-    result.success?
   end
 
-  banner.("SUMMARY: in #{'%.1fs' % total_time}\n\t#{summary.join("\n\t")}")
+  desc 'Report code coverage to CodeClimate'
```

Comment:
By documenting those tasks you are making them the public API that is exposed to the CLI.

This is similar to the `run_tests` task.  As mentioned in 
https://github.com/3scale/porta/pull/25/files#r222085667 one option of splitting Rake tasks is a Class/Module. IMO that is something we should be taking here. Creating tasks for splitting long tasks internally is a quite bad pattern in rake.

Reply:
i think this is now also addressed. thanks!